dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -36,4 +36,15 @@ def confirm_email
     end
   end
 
+  def undelete_user
+    user = User.only_deleted.find_by_id(params[:user_id])
+    if user
+      user.deleted_at = nil
```

Comment:
Though I initially was a fan of the suggestion, looking further, calling `user.restore` bypasses model validations and callbacks (see https://github.com/rubysherpas/paranoia/blob/9e0ec38e5f4d08d25c1bf1b63c1fe78208abe625/lib/paranoia.rb#L112). It seems worth running the validations. Any disagreement?